kernel/src/process_checker/basic.rs

```diff
@@ -31,6 +31,12 @@ pub struct AppCheckerSimulated<'a> {
     binary: OptionalCell<&'a [u8]>,
 }
 
+impl<'a> Default for AppCheckerSimulated<'a> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
That would be `cargo clippy --fix`

`#[derive(Default)]` works instead

Reply:
I'm not able to replicate this. If I remove these derive defaults, and run `make clippy` I don't get any errors/warnings.

In general, it's a bit of a subtle thing to add blanket Default implementations to a bunch of things. If they are implemented as wrappers around other public constructors, that's _probably_ fine, since it's already been deemed appropriate to allow assigning those values to private fields.

But more likely, I think we'd prefer to disable whatever new clippy rule is there and revert these changes unless we think carefully about each Default implementation.

Except I can't figure out which one it is

Reply:
Removing the `#[derive(Default)]` passes the checks, but `cargo clippy --fix` will re-add them

Reply:
I think you must have run `cargo clippy --fix` rather than using `tools/run_clippy.sh` (which is what `make clippy` uses), which includes a bunch of explicit changes for Tock of clippy's default lints.

Reply:
Ah! That explains it. Clippy is happy now